### DISCUSSION TITLE
Setting days during alarm configuration - CLOSED

### DISCUSSION THREAD

**Discussion by KaeroDot** [astlouys/Pico-Green-Clock#12] [Q&A] [answered]
Hi. I got a brand new green clock hardware and Pico W. I have uploaded astlouys firmware (v.8.0). All seems to be working but the alarm days settings. I can make a long press of the top button to start alarm configuration, continue through settings, switch on/off alarm, change hour and minutes of the alarm. Than I reach Alarm Days Setting. I see two alternatively blinking days, Monday and Friday. Using short press of Up or Down button, I can change target day in such way that I see alternatively blinking e.g. Wednesday and Friday. However long press of Up or Down button does nothing. I make long press Up on Wednesday, move to Thurdsay and the Wednesday is not switched on. All I can see is alternatively blinking Thursday and Friday.  Long press Down button does nothing also.
All other functions works as expected, setting time, setting F/C, DST, scrolling, hour chimes. I tried to upload firmware again,  recompile the firmware and upload, no change.
Does anyone got idea what can be wrong? I tried long press, longer press, nothing works.
Thanks for help.

**Comment by astlouys** [comment]
Hi,

Let me take a look... I assume you followed the instructions in the User Guide for your tests / tries ?

**Comment by KaeroDot** [comment]
Hi, yes, I have followed your User Guide. I have red all parts of the guide but the IR remote control. I am sure the hardware buttons works because I was able to set other things of the clock. Settings is properly saved into memory, tested by switching power of and on.
I am not sure if I should see two alternately blinking days when entering Alarm days settings? From the guide I understood only one day should be blinking, correct?

**Comment by astlouys** [comment]
Hi KaeroDot (your name is not indicated on you nick... ?),

You're right, something is wrong in the code. I've already checked it and it is a side-effect of an optimization I've done shortly before releasing version 8.00 about the "top indicators". In fact, the "logic" behind alarm setting is Ok, but blinking affects all indicators instead of only the "target day".

I'm almost done fixing the problem, but I prefer taking some time to test. I should be able to release a fix (Version 8.01) in a few hours.

May I ask you in which country you live ?  Did you have a chance to make some tests with "Daylight Saving Time" ("Summer Time") ?

André

**Reply by KaeroDot** [reply]
Good!
I live in Czech Republic. I have not yet made any DST tests. We use standard European DST changing. I can test it later if needed.
I would like to thank you for the code, you have invested a lot of time into it and it is marvellous. The guide is also wonderful think.

**Comment by astlouys** [comment]
I KaeroDot,

I'm happy to know that the software runs in Czech Republic... and thank you for your good words.

I just released Version 8.01 which should solve your problem with Alarm Setting. Could you give it a try and let me know how it goes... and let me know how is your experience with the Green Clock.

André

**Comment by KaeroDot** [comment]
It works. As you said, the logic was probably working, because several days were switched on, only the display was incorrect. Thanks!

**Reply by astlouys** [reply]
I'm glad it solved the problem.

Thanks to you and let me know how it goes as you use it...

**Comment by KaeroDot** [comment]
Oh, I am sorry I was too hasty. The alarm does not make any sound. At time of the alarm, the display just flickers very quickly from and to a clock. No sound is generated. It seems word "Alarm" flickers.
The key click works (BP: ON), so hardware is ok. I tried uploading older firmware 6.0, and the alarm worked and sound was generated as expected.

**Reply by KaeroDot** [reply]
And I forgot to add: Hour chime works, sound is generated.

**Reply by astlouys** [reply]
KaeroDot,

You're right, I was able to reproduce the problem. I'll check that... Keep an eye on a potential updated Version to fix it.

...and thank you for reporting those problems that you find !

**Reply by astlouys** [reply]
Version 8.02 should solve the problem you had with alarm not sounding. The problem was still related to days-of-week. In short, there were two different logics with days-of-week throughout the code. Given that, may I ask you to keep an eye on handling of day-of-week (within alarms, but also elsewhere in the code) and let me know if ever you see some strange behaviour.

Thanks.

**Comment by KaeroDot** [comment, marked as answer]
Hi, the alarm now works, I have tested it for a week. Thanks for fixing it.
I have also tested DST. I have set value of ST to 5 (European Union), set time to 1:00, date to 26. 3. 2023 and waited if the time will change from 2:00 to 3:00. Unfortunately the clock have not changed. TZ was set to 1.

**Reply by astlouys** [reply]
Hi, thanks for reporting it.

Good news first: I'm looking at it and it should be solved in the next version (Version 9.00) that should be released in the coming week (more or less).

Bad news now: Version 9.00 will also include NTP ("Network Time Protocol") to synchronize the Green Clock with an Internet time-reference server (if you use a Pico W, which you said before). I wouldn't be surprised that NTP may have some impact on DST and Timezone. Also, I use to have complete control over the Pico in the clock for my tests, but it is another ball game with NTP. I can't modify the time from the servers and have to do with it... So, we may have to revisit DST in March if ever some more glitch appears...

In the meantime, let me know how it goes with Version 9.00 when it is released.

**Reply by KaeroDot** [reply]
Thanks, NTP is a good news! I will test it when ready.

**Reply by astlouys** [reply]
Hi KaeroDot,

The Version 9.00 of the Green Clock Firmware is now available. I made changes (optimizations and bug fixing) to Summer Time handling and it should be OK now. That being said, and as I mentioned before, NTP implementation may have side effects on it. I will keep an eye on it and do not hesitate to let me know if you see any strange behaviour on your side. As I wrote in another thread, Summer Time testing becomes more difficult now since NTP restore local time relatively fast (and turning Off NTP doesn't let us know if it has an impact on Summer Time... chicken-and-egg type of thing...)

I hope you'll enjoy that new version.

André
